source/PlayerInfo.cpp

```diff
@@ -1356,7 +1367,19 @@ void PlayerInfo::SetShipOrder(const vector<shared_ptr<Ship>> &newOrder)
 	// Check if the incoming vector contains the same elements
 	if(std::is_permutation(ships.begin(), ships.end(), newOrder.begin()))
 	{
+		shared_ptr<Ship> oldFirstShip = ships.front();
 		ships = newOrder;
+		shared_ptr<Ship> newFirstShip = ships.front();
+		// Check if the position of the flagship has changed and
+		// if the ship in the first position can be a flagship and is in the current system.
+		if(newFirstShip != oldFirstShip && Preferences::Has("Automatically unpark flagship"))
+		{
+			if(newFirstShip->CanBeFlagship() && newFirstShip->GetSystem() == system)
```

Comment:
Why two separate ifs?

Reply:
Oh yee, as we deleted the pos i actually dont need a second if case